### ISSUE TITLE
fix: Could not open cp_proj generic class cache for build file

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#1393]
+1 

`flutter create my_app`
`shorebird init`

````
❯ shorebird init
✗ Detecting product flavors (6.7s)
Unable to extract product flavors.
Exception: Starting a Gradle Daemon, 3 busy and 5 incompatible and 1 stopped Daemons could not be reused, use --status for details


FAILURE: Build failed with an exception.

* Where:
Build file '\my_app\android\build.gradle' line: 26

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not open cp_proj generic class cache for build file \my_app\android\app\build.gradle' (.gradle\caches\7.5\scripts\52c2dfmq86br7sqw5y13kinam).
      > BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 63

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
````

**flutter doctor**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.7, on Microsoft Windows [Version 10.0.22621.2361], locale en-US)
[✓] Windows Version (Installed version of Windows is version 10 or higher)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Chrome - develop for the web
[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.5.4)
[✓] Android Studio (version 2022.3)
[✓] IntelliJ IDEA Community Edition (version 2023.1)
[✓] VS Code (version 1.83.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

**shorebird doctor**
```
Shorebird 0.16.1 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.13.6 • revision ef4b661ddc0c71b738432ae59c6bc573e917854b
Engine • revision 35809a641ed60f9595fcb6c6f939a8768eaca976

✓ Shorebird is up-to-date (0.7s)
✓ Flutter install is correct (0.7s)
✓ AndroidManifest.xml files contain INTERNET permission (44ms)

No issues detected!
```

_Originally posted by @C4co in https://github.com/shorebirdtech/shorebird/issues/1392#issuecomment-1759747252_

**Comment by eseidel** [comment]
Hmm, this might be https://stackoverflow.com/questions/32513740/gradle-build-failure-could-not-open-proj-class-cache-for-build-file?

Can you paste the output of `shorebird doctor -v`?  That will include the java version.

**Comment by eseidel** [comment]
It may also be you just need to clear your gradle cache that it may have been corrupted somehow?

**Comment by eseidel** [comment]
FYI @C4co in case it didn't CC you on the issue (I've never used the "create new issue from comment" feature in GitHub before this morning).

**Comment by carllosnc** [comment]
@eseidel 

```
❯ shorebird doctor -v
Shorebird 0.16.1 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.13.6 • revision ef4b661ddc0c71b738432ae59c6bc573e917854b
Engine • revision 35809a641ed60f9595fcb6c6f939a8768eaca976

Android Toolchain
  • Android Studio: C:\Program Files\Android\Android Studio
  • Android SDK: C:\Users\carll\AppData\Local\Android\sdk
  • ADB: C:\Users\carll\AppData\Local\Android\sdk\platform-tools\adb.exe
  • JAVA_HOME: C:\Program Files\Java\jdk-19

✓ Shorebird is up-to-date (0.7s)
✓ Flutter install is correct (0.6s)
✓ AndroidManifest.xml files contain INTERNET permission (44ms)

No issues detected!
```

I cleared the cache but still had the same problem. But it worked when I ran `gradle wrapper` inside the android folder.

**Comment by eseidel** [comment]
My guess is it may be the java version.  I'd try unsetting JAVA_HOME and see if it works.  e.g. https://github.com/shorebirdtech/shorebird/issues/1163

`unset JAVA_HOME` in your shell before you run might work?

**Comment by carllosnc** [comment]
> My guess is it may be the java version. I'd try unsetting JAVA_HOME and see if it works. e.g. #1163
> 
> `unset JAVA_HOME` in your shell before you run might work?


❯ shorebird doctor -v
Shorebird 0.16.1 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.13.6 • revision ef4b661ddc0c71b738432ae59c6bc573e917854b
Engine • revision 35809a641ed60f9595fcb6c6f939a8768eaca976

Android Toolchain
  • Android Studio: C:\Program Files\Android\Android Studio
  • Android SDK: C:\Users\carll\AppData\Local\Android\sdk
  • ADB: C:\Users\carll\AppData\Local\Android\sdk\platform-tools\adb.exe
  • **JAVA_HOME: C:\Program Files\Android\Android Studio\jbr**

✓ Shorebird is up-to-date (0.7s)
✓ Flutter install is correct (0.7s)
✓ AndroidManifest.xml files contain INTERNET permission (45ms)

No issues detected!

---

Worked. This was the problem.

**Comment by eseidel** [comment]
Thanks.  So this is yet another duplicate of #1163.  We'll likely look at that next week.  Thank you!

**Comment by SamuelAc1011** [comment]
I have the same problem

But I can't solve the android studio being detected and the java home location being correct

```
PS C:\projects\try_shorebird> shorebird doctor -v
Shorebird 0.18.2 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.13.9 • revision 39df2792f537b1fc62a9c668a6990f585bd91456
Engine • revision e81fa131e59506d9f6af2a0cee7de749131f1bf0

Android Toolchain
  • Android Studio: not detected
  • Android SDK: E:\Users\Sam\AppData\Local\Android\Sdk
  • ADB: E:\Users\Sam\AppData\Local\Android\Sdk\platform-tools\adb.exe
  • JAVA_HOME: C:\Android Studio\jbr\bin\java.exe

✓ Shorebird is up-to-date (0.6s)
✓ Flutter install is correct (0.6s)
✓ AndroidManifest.xml files contain INTERNET permission (37ms)
✓ Has access to storage.googleapis.com (34ms)

No issues detected!
```

**Comment by eseidel** [comment]
> I have the same problem
> 
> But I can't solve the android studio being detected and the java home location being correct
>
> Android Toolchain
>   • Android Studio: not detected
>   • Android SDK: E:\Users\Sam\AppData\Local\Android\Sdk
>   • ADB: E:\Users\Sam\AppData\Local\Android\Sdk\platform-tools\adb.exe
>   • JAVA_HOME: C:\Android Studio\jbr\bin\java.exe

That looks like it's using java from one place and Android from another, which may not go well.  I suspect we still need to improve our java detection logic.

**Comment by eseidel** [comment]
@SamuelAc1011 does `shorebird` work correctly for you with that setup?  If not, would you be willing to open a new issue so your comments don't get lost on this closed one?  Thanks!

**Comment by 72worthyTh** [comment]
i have the same issue

S C:\flutterProjects\llutterfirebase\android> ./gradlew signingReport

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\flutterProjects\llutterfirebase\android\build.gradle' line: 26

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not open cp_proj generic class cache for build file 'C:\flutterProjects\llutterfirebase\android\app\build.gradle' (C:\Users\INFONET\.gradle\caches\7.5\scripts\cfb827u4qxbb1k7hy3dicjnjd).
      > BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 65

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s

**Comment by amitkumarcoding** [comment]
I resolved the issue by following these steps:

1. Open Android Studio.
2. Drag and drop the android folder of your project into Android Studio.
3. Wait for the build process to complete.
4. Once the process is finished, everything should be working correctly.

**Comment by Moinuddin9777** [comment]
For me this very issue came up after migrating an old project, I had no implementation of shorebird in that project.

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Could not open cp_proj generic class cache for build file '/Users/username/projects/project_name/android/app/build.gradle' (/Users/username/.gradle/caches/7.3.3/scripts/euzwbzznt4f7dpdx1y8ys6uaz).
   > BUG! exception in phase 'conversion' in source unit '_BuildScript_' class org.codehaus.groovy.ast.expr.TupleExpression cannot be cast to class org.codehaus.groovy.ast.expr.ArgumentListExpression (org.codehaus.groovy.ast.expr.TupleExpression and org.codehaus.groovy.ast.expr.ArgumentListExpression are in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @6767c1fc)
   
   ```
   
   What all I tried:

- flutter clean then flutter pub get
- flutter clean then flutter . create
- migrating settings.gradle, build.gradle, app/build.gradle as per documentation
- commented out sub projects
- deleted the whole .gradle file which has 7.3.3 as AGP 
- re built the project which created the .gradle with 7.3.3 AGP
- removed and replaced the .gradle folder with a new one from a newly created project which had 7.6.0 AGP
- flutter clean and flutter pub get
- stashed gradle changes and run the project
- rm rf .gradle/caches and rebuild (just now)   

   
   In my case it might be due to deprecation of some files in .gradle
